src/Stratis.Bitcoin.Tests/Consensus/ChainedHeaderTreeTest.cs

```diff
@@ -150,5 +155,96 @@ public void ConnectHeaders_NewAndExistingHeaders_ShouldCreateNewHeaders()
+            cht.ConnectNewHeaders(1, listOfExistingHeaders);
+            var afterKeys = new List<uint256>(cht.GetChainedHeadersByHash().Keys);
+
+            // check chainedHeadersByHash map before and after adding duplicate headers
+            Assert.Equal(beforeKeys.Count, afterKeys.Count);
+            Assert.True(beforeKeys.All(afterKeys.Contains));
+        }
+
+        // Supply some headers and after that supply some more headers
+        // and make sure that PeerTipsByHash is updated (the total amount of items is the same).
+        [Fact]
+        public void ConnectHeaders_SupplyHeadersThenSupplyMore_PeerTipsByHashShouldBeUpdated()
+        {
+            TestContext testContext = new TestContext();
+            ChainedHeaderTree cht = testContext.CreateChainedHeaderTree();
+
+            var chainTip = testContext.ExtendAChain(10);
+            cht.Initialize(chainTip, true);
+            chainTip.BlockDataAvailability = BlockDataAvailabilityState.BlockAvailable;
+            ChainedHeader newChainTip = testContext.ExtendAChain(10, chainTip);
+
+            var listOfExistingHeaders = testContext.ChainedHeaderToList(chainTip, 10);
+            var listOfNewHeaders = testContext.ChainedHeaderToList(newChainTip, 10);
+
+            testContext.ChainStateMock.Setup(s => s.ConsensusTip).Returns(chainTip);
+            chainTip.BlockValidationState = ValidationState.FullyValidated;
+
+            cht.ConnectNewHeaders(1, listOfExistingHeaders);
+
+            var peerIdsByTipHashBefore = new Dictionary<uint256, HashSet<int>>(cht.GetPeerIdsByTipHash());
+            var chainedHeaderTreeHashesBefore = cht.GetChainedHeadersByHash();
+
+            cht.ConnectNewHeaders(1, listOfNewHeaders);
+
+            var peerIdsByTipHashAfter = cht.GetPeerIdsByTipHash();
+            var chainedHeaderTreeHashesAfter = cht.GetChainedHeadersByHash();
+
+            // Same number of entries as we are reassigning hash
+            Assert.True(chainedHeaderTreeHashesBefore.Count == chainedHeaderTreeHashesAfter.Count );
+
+            // Peer's tip hash has changed
+            Assert.True(peerIdsByTipHashBefore.FirstOrDefault(x => x.Value.Contains(1)).Key !=
+                        peerIdsByTipHashAfter.FirstOrDefault(x => x.Value.Contains(1)).Key);
+        }
+
+        // Make sure checkpoints are off - supply some headers,
+        // CHT should return ToDownload array of the same size as the amount of headers.
+        [Fact]
+        public void ConnectHeaders_SupplyHeaders_ToDownloadArraySizeSameAsNumberOfHeaders()
+        {
+            // Setup
+            var ctx = new TestContext();
+            ChainedHeaderTree cht = ctx.CreateChainedHeaderTree();
+            var chainTip = ctx.ExtendAChain(10);
+            cht.Initialize(chainTip, true);
+            chainTip.BlockDataAvailability = BlockDataAvailabilityState.BlockAvailable;
+            ctx.ChainStateMock.Setup(s => s.ConsensusTip).Returns(chainTip);
```

Comment:
Do we really need to make this setup? Is it used anywhere?

Reply:
Yes I need both the CHT and the context (for assigning and checking checkpoints)

Reply:
i meant specifically this ConsensusTip mock, where is it used?

Reply:
Nowhere.  I've removed all instances.  You are correct - I think this was required by earlier code but is now redundant.